### DISCUSSION TITLE
Reference implementation divergence

### DISCUSSION THREAD

**Discussion by marcospassos** [ariakit/ariakit#1522] [Questions] [answered]
The W3C combobox's reference implementation sets the `aria-activedescendant` to the display element instead of the listbox, as is the case in the [Ariakit's select example](https://ariakit.org/examples/select). 

Also, the reference doesn't focus on the popup when opening the select box. In fact, this behavior is explicitly defined as:

> Opens the listbox without moving focus or changing selection.

But the example also diverges on this point:
![image](https://user-images.githubusercontent.com/943036/174449093-cb865aa6-62e3-4aee-99c9-7dd839cf0da3.png)

What behavior is correct?

**Comment by marcospassos** [comment]
Another issue I've found is that screen readers announce the input label every time you move the selection on the list. You can see it happening [here](https://ariakit.org/examples/combobox-matches) (it affects all combobox examples).

Disabling virtual focus fix the issue, but by doing this, the implementation does not follow the W3C recommendation anymore: the `aria-activedescendant` no longer is set, and the input loses the focus. [In the official](https://w3c.github.io/aria-practices/examples/combobox/combobox-autocomplete-none.html) example, the behavior follows the letter.

**Reply by diegohaz** [reply]
It looks like a separate issue. I couldn't reproduce it with VoiceOver/Safari and NVDA/Chrome. Could you submit an issue with the software combination? A video will also help.

**Reply by diegohaz** [reply]
> Disabling virtual focus fix the issue, but by doing this, the implementation does not adhere to the W3C specification anymore

The only required attribute on `role="combobox"` elements is `aria-expanded`. `aria-activedescendant` is not required. The combobox with `virtualFocus: false` is totally valid.

**Reply by marcospassos** [reply]
Hi @diegohaz, here is the video that shows the issue. I'm using [this Chrome's extension (v53.0.2784.130)](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn)

https://user-images.githubusercontent.com/943036/174612577-6e872469-7299-4496-aa9d-afb50a64756c.mov

(turn the sound on)

Notice in the first example that the labels are repeated on every change, while in the select-combobox example, the input label is reannounced every time.

**Reply by diegohaz** [reply]
I see. That's probably a bug in that Chrome extension (that's in maintenance mode). We don't support that.

**Reply by marcospassos** [reply]
I thought that in the first place, but the same doesn't happen in the W3C combobox example.

What are the screenreaders officially supported?

**Reply by diegohaz** [reply]
NVDA, JAWS, VoiceOver and TalkBack.

**Comment by diegohaz** [comment, marked as answer]
Yes, the Select differs from the ARIA Authoring Practices Guide (APG) in the focus aspect. It's important to note that the APG is not a specification. It's a recommendation. The Select pattern is even more interesting as it doesn't have a dedicated role type. Historically, it's been implemented with the disclosure pattern plus a listbox, where the focus remains on the listbox.

After testing several browser and screen reader combinations, the version where the listbox receives focus worked better. For example, rendering the select as `modal` is a valid use case (in fact, native selects have a modal state). Focusing on an element outside the popover while in a modal state turned out to be a terrible user experience when using a screen-reader. It needed several workarounds, whereas focusing on the listbox works without caveats.

**Reply by marcospassos** [reply]
Thanks for clarifying.

I've also noticed that multi-select comboboxes change the focus to the item whenever you select it, which is different from how the single-select combobox behaves. Can you confirm if it is the expected behavior? If not, I can open a separate issue.

**Reply by diegohaz** [reply]
That's expected. That example has `virtualFocus` set to false (there's an explanation in the code).
